<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ECGDialog</class>
 <widget class="QDialog" name="ECGDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1096</width>
    <height>485</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>ECG Simulation</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_7" columnstretch="0,0,0,0,1,0,0,0">
   <item row="0" column="0">
    <widget class="QScrollArea" name="scrollArea">
     <property name="minimumSize">
      <size>
       <width>394</width>
       <height>0</height>
      </size>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>-608</y>
        <width>387</width>
        <height>900</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>P Wave Parameters</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="p_mag_label">
            <property name="text">
             <string>Magnitude</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="p_wid_label">
            <property name="text">
             <string>Widthness</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="p_time_labe">
            <property name="text">
             <string>Time</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QDoubleSpinBox" name="p_mag_double_spinbox">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>0.300000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.180000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="p_wid_double_spinbox">
            <property name="decimals">
             <number>5</number>
            </property>
            <property name="minimum">
             <double>0.010000000000000</double>
            </property>
            <property name="maximum">
             <double>0.030000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.000100000000000</double>
            </property>
            <property name="value">
             <double>0.017800000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QDoubleSpinBox" name="p_time_double_spinbox">
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.200000000000000</double>
            </property>
            <property name="maximum">
             <double>0.300000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.000100000000000</double>
            </property>
            <property name="value">
             <double>0.239800000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Q Wave Parameters</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="q1_mag_label">
            <property name="text">
             <string>Q1 Magnitude</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QLabel" name="q1_wid_label">
            <property name="text">
             <string>Q1 Widthness</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3" colspan="2">
           <widget class="QLabel" name="q1_time_labe">
            <property name="text">
             <string>Q1 Time</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QDoubleSpinBox" name="q1_mag_double_spinbox">
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>-0.200000000000000</double>
            </property>
            <property name="maximum">
             <double>-0.100000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>-0.110300000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="QDoubleSpinBox" name="q1_wid_double_spinbox">
            <property name="decimals">
             <number>5</number>
            </property>
            <property name="minimum">
             <double>0.025000000000000</double>
            </property>
            <property name="maximum">
             <double>0.035000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.030640000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="3" colspan="2">
           <widget class="QDoubleSpinBox" name="q1_time_double_spinbox">
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.250000000000000</double>
            </property>
            <property name="maximum">
             <double>0.350000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.321800000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QLabel" name="q2_mag_label">
            <property name="text">
             <string>Q2 Magnitude</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2" colspan="2">
           <widget class="QLabel" name="q2_wid_label">
            <property name="text">
             <string>Q2 Widthness</string>
            </property>
           </widget>
          </item>
          <item row="2" column="4">
           <widget class="QLabel" name="q2_time_label">
            <property name="text">
             <string>Q2 Time</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QDoubleSpinBox" name="q2_mag_double_spinbox">
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>-0.150000000000000</double>
            </property>
            <property name="maximum">
             <double>-0.050000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>-0.107500000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="2" colspan="2">
           <widget class="QDoubleSpinBox" name="q2_wid_double_spinbox">
            <property name="decimals">
             <number>5</number>
            </property>
            <property name="minimum">
             <double>0.005000000000000</double>
            </property>
            <property name="maximum">
             <double>0.006000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.000100000000000</double>
            </property>
            <property name="value">
             <double>0.005710000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="4">
           <widget class="QDoubleSpinBox" name="q2_time_double_spinbox">
            <property name="decimals">
             <number>5</number>
            </property>
            <property name="minimum">
             <double>0.360000000000000</double>
            </property>
            <property name="maximum">
             <double>0.380000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.371230000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>R Wave Parameters</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="r_mag_label">
            <property name="text">
             <string>Magnitude</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="r_wid_label">
            <property name="text">
             <string>Widthness</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="r_time_label">
            <property name="text">
             <string>Time</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QDoubleSpinBox" name="r_mag_double_spinbox">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>0.025000000000000</double>
            </property>
            <property name="maximum">
             <double>0.125000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.050000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="r_wid_double_spinbox">
            <property name="decimals">
             <number>5</number>
            </property>
            <property name="minimum">
             <double>0.029000000000000</double>
            </property>
            <property name="maximum">
             <double>0.031000000000000</double>
            </property>
            <property name="value">
             <double>0.029870000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QDoubleSpinBox" name="r_time_double_spinbox">
            <property name="decimals">
             <number>5</number>
            </property>
            <property name="minimum">
             <double>0.460000000000000</double>
            </property>
            <property name="maximum">
             <double>0.470000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.465710000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>S Wave Parameters</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="s_mag_label_2">
            <property name="text">
             <string>Magnitude</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="s_wid_label">
            <property name="text">
             <string>Widthness</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="s_time_label">
            <property name="text">
             <string>Time</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QDoubleSpinBox" name="s_mag_double_spinbox">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>0.400000000000000</double>
            </property>
            <property name="maximum">
             <double>0.600000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.509000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="s_wid_double_spinbox">
            <property name="decimals">
             <number>5</number>
            </property>
            <property name="minimum">
             <double>0.008000000000000</double>
            </property>
            <property name="maximum">
             <double>0.010000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.000100000000000</double>
            </property>
            <property name="value">
             <double>0.009090000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QDoubleSpinBox" name="s_time_double_spinbox">
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.460000000000000</double>
            </property>
            <property name="maximum">
             <double>0.480000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.476900000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>T Wave Parameters</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="2">
           <widget class="QLabel" name="t_time_label">
            <property name="text">
             <string>Time</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QDoubleSpinBox" name="t_mag_double_spinbox">
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.310000000000000</double>
            </property>
            <property name="maximum">
             <double>0.330000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.325500000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="t_wid_label">
            <property name="text">
             <string>Widthness</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="t_wid_double_spinbox">
            <property name="decimals">
             <number>5</number>
            </property>
            <property name="minimum">
             <double>0.028000000000000</double>
            </property>
            <property name="maximum">
             <double>0.031000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.000100000000000</double>
            </property>
            <property name="value">
             <double>0.029780000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QDoubleSpinBox" name="t_time_double_spinbox">
            <property name="decimals">
             <number>4</number>
            </property>
            <property name="minimum">
             <double>0.740000000000000</double>
            </property>
            <property name="maximum">
             <double>0.760000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="t_mag_label">
            <property name="text">
             <string>Magnitude</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_6">
         <property name="title">
          <string>Signal Properties</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="10" column="0">
           <widget class="QLabel" name="period_label">
            <property name="text">
             <string>Period (s)</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="noise_label">
            <property name="text">
             <string>Noise (Mean Value)</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLineEdit" name="simulation_line_edit">
            <property name="placeholderText">
             <string>ECG Simulation</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="duration_label">
            <property name="text">
             <string>Duration (s)</string>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QDoubleSpinBox" name="duration_spinbox">
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>21600.000000000000000</double>
            </property>
            <property name="value">
             <double>10.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QDoubleSpinBox" name="noise_double_spinbox">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="maximum">
             <double>100.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="sampling_frequency_label">
            <property name="text">
             <string>Sampling Frequency</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="signal_name_label">
            <property name="text">
             <string>Signal Name</string>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QDoubleSpinBox" name="period_spinbox">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>5.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.900000000000000</double>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QSpinBox" name="sampling_frequency_spinbox">
            <property name="minimum">
             <number>100</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="value">
             <number>256</number>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="delay_label">
            <property name="text">
             <string>Delay (s)</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QDoubleSpinBox" name="delay_spin_box">
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="0" column="1" colspan="7">
    <widget class="PlotWidget" name="plot"/>
   </item>
   <item row="3" column="0">
    <widget class="QSlider" name="scale_slider">
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>200</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="pageStep">
      <number>10</number>
     </property>
     <property name="sliderPosition">
      <number>100</number>
     </property>
     <property name="tracking">
      <bool>true</bool>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::TicksBelow</enum>
     </property>
     <property name="tickInterval">
      <number>50</number>
     </property>
    </widget>
   </item>
   <item row="3" column="6">
    <widget class="QPushButton" name="reset_signal_button">
     <property name="text">
      <string>Reset Signal</string>
     </property>
     <property name="autoDefault">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="scale_value">
     <property name="text">
      <string>Scale: 1</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="8">
    <widget class="PlotWidget" name="main_plot"/>
   </item>
   <item row="3" column="5">
    <widget class="QPushButton" name="create_signal_button">
     <property name="text">
      <string>Create Signal</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header location="global">pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
